--- a/Makefile.in
+++ b/Makefile.in
@@ -152,6 +152,10 @@ INCLS = -I. @V_INCLS@ ${PCAP_INCS}
 # FreeBSD needs	-Ae
 # HP needs	-Ae
 CFLAGS = $(CCOPT) $(DEFINES) @DEFS@ @V_DEFINES@ $(INCLS)
+CFLAGS += -g -Wall -O2
+
+# Define _BSD_SOURCE so that the right tcp header structures are found
+CFLAGS += -D_BSD_SOURCE
 
 # Standard LIBS
 LDLIBS = @LIBS@ ${PCAP_LDLIBS}
@@ -210,26 +214,11 @@ YACC_VAL=@YACC@
 LEX_VAL=@LEX@
 
 filt_parser.c: filt_parser.y filter.h
-	if test "${YACC_VAL}" = "bison -y" ; then \
-		@YACC@ -vd -p filtyy -o filt_parser.c $(srcdir)/filt_parser.y ;\
-		cp filt_parser.c flex_bison ;\
-		cp filt_parser.h flex_bison ;\
-	else \
-		echo "Could not find BISON on this system";\
-		echo "Copying the BISON output files generated at our place" ;\
-		cp flex_bison/filt_parser.c . ;\
-		cp flex_bison/filt_parser.h . ;\
-	fi
+	cp flex_bison/filt_parser.c .
+	cp flex_bison/filt_parser.h .
 
 filt_scanner.c: filt_scanner.l filter.h filt_parser.h
-	if test ${LEX_VAL} = "flex" ; then \
-		@LEX@ -t -Pfiltyy $(srcdir)/filt_scanner.l > filt_scanner.c ;\
-		cp filt_scanner.c flex_bison ;\
-	else \
-		echo "Could not find FLEX on this system" ;\
-		echo "Copying the FLEX output files generated at our place" ;\
-		cp flex_bison/filt_scanner.c . ;\
-	fi
+	cp flex_bison/filt_scanner.c .
 
 # filt_parser.h created as a side effect of running yacc...
 filt_parser.h: filt_parser.c
@@ -286,6 +275,9 @@ bigtarfile:
 #
 # just a quick installation rule
 INSTALL = ./install-sh -c
+ifeq (,$(findstring nostrip,$(DEB_BUILD_OPTIONS)))
+INSTALL += -s
+endif
 install: tcptrace install-man
 	$(INSTALL) -m 755 -o bin -g bin tcptrace ${BINDIR}/tcptrace
 install-man: 
